extensions/data-plane-selector/data-plane-selector-client/src/main/java/org/eclipse/edc/connector/dataplane/selector/RemoteDataPlaneSelectorService.java

```diff
@@ -123,38 +124,52 @@ public DataPlaneInstance select(DataAddress source, DataAddress destination, Str
 
     @Override
     public ServiceResult<Void> addInstance(DataPlaneInstance instance) {
-        throw new UnsupportedOperationException("not implemented");
-    }
+        var jsonObject = typeTransformerRegistry.transform(instance, JsonObject.class).getContent();
 
+        var requestBody = Json.createObjectBuilder(jsonObject)
+                .add(CONTEXT, createObjectBuilder().add(VOCAB, EDC_NAMESPACE))
+                .build();
+        var body = RequestBody.create(requestBody.toString(), TYPE_JSON);
 
-    private <R> R handleResponse(Response response, TypeReference<? extends R> tr, R defaultValue) {
-        if (response.isSuccessful()) {
-            R r = handleSuccess(response, tr);
-            return r == null ? defaultValue : r;
-        } else {
-            return handleError(response);
-        }
-    }
+        var request = new Request.Builder().post(body).url(url).build();
 
-    private <R> R handleSuccess(Response response, TypeReference<R> tr) {
         try {
-            var body = response.body().string();
-            if (StringUtils.isNullOrEmpty(body)) {
-                return null;
+            try (var response = httpClient.execute(request)) {
+
+                handleResponse(response, null, null);
+                return ServiceResult.success();
             }
+        } catch (IOException e) {
+            return ServiceResult.badRequest(e.getMessage());
+        }
+    }
 
-            return mapper.readValue(body, tr);
+    private <R> R handleResponse(Response response, TypeReference<? extends R> tr, R defaultValue) {
```

Comment:
how about returning a `Result<R>` here and evaluating that in client code, e.g. the `addInstance` method?

Reply:
totally agree, given the size of the PR I added that to my TODO list, but let's see if doing that will help me solve the other 2 comments